ZEngine/src/VulkanDevice.cpp

```diff
@@ -152,63 +162,140 @@ namespace ZEngine::Hardwares
         return m_logical_device;
     }
 
-    VkQueue VulkanDevice::GetCurrentGraphicQueue(bool with_present_support) const
+    QueueView VulkanDevice::GetAnyGraphicQueue() const
     {
-        if (with_present_support)
-        {
-            assert(m_graphic_with_present_support_queue != VK_NULL_HANDLE);
-            return m_graphic_with_present_support_queue;
-        }
-        assert(m_graphic_queue != VK_NULL_HANDLE);
-        return m_graphic_queue;
+        ZENGINE_VALIDATE_ASSERT(!m_graphic_with_present_support_queue_collection.empty(), "Graphic Queue with Present support can't be empty")
```

Comment:
It looks to me like your intent is to continously select a queue at random and submit different chunks of graphics work to them. This is not really how you should achieve parallelism in Vulkan. 

Like this, you're going to run into complicated synchronization issues very quickly, for example if you're having multiple random queues executing workloads of the same scene into the same framebuffers in parallel. Also, you're going to run into ordering issues of pixel writes. All synchronization has to be done manually in Vulkan, meaning the synchronization between queues as well. So if you're not expecting to have completely independent graphics workloads, multiple random queues will just make everything needlessly complicated.

Reply:
thanks a lot for the feedback.
I revisited the whole engine resource management to stick with only needed queue (transfer, graphic)  and maybe later compute